clients/githubrepo/languages.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2021 Security Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package githubrepo
+
+import (
+	"context"
+	"fmt"
+	"path"
+	"sync"
+
+	"github.com/google/go-github/v38/github"
+)
+
+type languagesHandler struct {
+	ghclient  *github.Client
+	once      *sync.Once
+	ctx       context.Context
+	errSetup  error
+	repourl   *repoURL
+	languages map[string]int
+}
+
+func (handler *languagesHandler) init(ctx context.Context, repourl *repoURL) {
+	handler.ctx = ctx
+	handler.repourl = repourl
+	handler.errSetup = nil
+	handler.once = new(sync.Once)
+}
+
+func (handler *languagesHandler) setup() error {
+	handler.once.Do(func() {
```

Comment:
one last thing before we merge. Can you create a structure for languages. You can add it as part of the interfaces in `./clients/`. It would be great to also have a `type Language string` and the list of languages listed there. This way the interface is not tied to GitHub

Reply:
np, working on it, will push the change very soon.